raiden_libs/blockchain.py

```diff
@@ -272,3 +225,61 @@ def _update(self):
 
         if not self.wait_sync_event.is_set() and new_unconfirmed_head_number == current_block:
             self.wait_sync_event.set()
+
+    def filter_events(self, filter_params: dict, name_to_callback: dict) -> None:
+        """Params:
+            filter_params: arguments for the filter call
+            name_to_callback: dict that maps event name to callbacks executed
+                if the event is emmited"""
+        for event_name, callback in name_to_callback.items():
+            events = get_events(
+                self.web3,
+                self.contract_manager,
+                self.contract_name,
+                event_name,
+                **filter_params
+            )
+            for event in events:
+                log.debug('Received confirmed %s event', event_name)
+                callback(event)
+
+    def reset_unconfirmed_on_reorg(self, current_block):
+        """Test if chain reorganization happened (head number used in previous pass is greater than
+        current_block parameter) and in that case reset unconfirmed event list."""
+        if self.wait_sync_event.is_set():  # but not on first sync
+            if current_block < self.unconfirmed_head_number:
```

Comment:
I believe the block number is not sufficient to detect reorgs, any plans to adjust and use the blockhash?

Reply:
Opened #25 to track that.